<?xml version="1.0" encoding="utf-8"?>

<launch>
	<!-- TODO remove anymal_config as it is not used -->
	<rosparam file="$(find quadruped_locomotion)/config/anymal_config.yaml" command="load"/>

	<include file="$(find gazebo_ros)/launch/empty_world.launch"/>

	<!--  Load anymal description into ros parameter server -->
	<include file="$(find anymal_c_simple_description)/launch/load.launch"/>

  <!-- Give the robot description parameter a name -->
  <arg name="description_name"           default="anymal_description"/>
  <!-- Joint state topic name -->
  <arg name="joint_states_topic"       default="/joint_states"/>

	<!-- Spawn anymal into Gazebo -->
	<node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model"
		args="-param anymal_description -urdf -model anymal -z 1 -R 0
			-J LF_HAA 0.0 -J LF_HFE 2.0 -J LF_KFE -2.5
			-J LH_HAA 0.0 -J LH_HFE -2.0 -J LH_KFE 2.5
			-J RF_HAA 0.0 -J RF_HFE 2.0 -J RF_KFE -2.5
			-J RH_HAA 0.0 -J RH_HFE -2.0 -J RH_KFE 2.5"/>

	<node pkg="quadruped_locomotion" type="joint_controller" output="screen" name="joint_controller"/>

	<node pkg="quadruped_locomotion" type="joint_state_publisher" output="screen" name="joint_state_publisher">
		<remap from="joint_positions" to="anymal/joint_positions"/>
		<remap from="joint_velocities" to="anymal/joint_velocities"/>
	</node>

  <!-- Robot state publisher -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" output="screen">
    <param name="publish_frequency" value="100"/>
    <param name="use_tf_static"     value="true"/>
    <remap from="robot_description" to="$(arg description_name)"/>
    <remap from="joint_states"      to="$(arg joint_states_topic)"/>
  </node>

	<node name="rviz" pkg="rviz" type="rviz"
		args="-d $(find quadruped_locomotion)/rviz/show_motion_plan.rviz"/>
</launch>
